src/components/Map/Map.js

```diff
@@ -1,23 +1,26 @@
 import './Map.css';
-import { useEffect } from 'react';
+import { useEffect, useState } from 'react';
 import { mapService } from '../../services/mapService';
+import { apiService } from '../../services/apiService';
 
 const Map = () => {
+  const [userPosition, setUserPosition] = useState({
+    latitude: 45,
+    longitude: 25
+  });
+
+  const [coffeeShops, setCoffeeShops] = useState([])
+
   useEffect(() => {
-    mapService.drawMap(
-      {
-        latitude: 45,
-        longitude: 25
-      },
-      [
-        {latitude: 50, longitude: 5},
-        {latitude: 40, longitude: -60},
-        {latitude: -30, longitude: -50},
-        {latitude: 0, longitude: 0},
-        {latitude: -80, longitude: 20}
-      ]
-    );
-  }, []);
+    async function validateAndFetchData(){      
+      if (apiService.validateValues(userPosition.latitude, userPosition.longitude)) {
+        setCoffeeShops(await apiService.getNearestShops(userPosition));
+      }
+    }
```

Comment:
Would be nice to have this function extracted outside of the `useEffect`, this makes the function reusable, and doesn't bloat the `useEffect`, making it easier to read.